src/libcore/sync/atomic.rs

```diff
@@ -310,11 +310,103 @@ impl AtomicBool {
     /// ```
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
+    #[rustc_deprecated(reason = "replaced with compare_exchange", since = "1.8.0")]
```

Comment:
I think we'll actually want to hold off on the deprecation for now, we'll likely want to take the road of:

1. Land the new methods as unstable
2. Stabilize the new methods (and *document* these as deprecated but not actually deprecate them)
3. Wait for the new methods to hit stable on the stable channel
4. Then we deprecate this method

A bit long winded but should help provide a smooth transition path!